docs/specs/protection.md

```diff
@@ -0,0 +1,34 @@
+# Protection
+
+This document specifies project-level protection configurations.
+
+- [IP Blocklist](#ip-blocklist)
+- [authgear.yaml](#authgearyaml)
+- [Error response](#error-response)
+
+## IP Blocklist
+
+### authgear.yaml
+
+```yaml
+protection:
```

Comment:
I found the keyword `protection` could be confusing because we already have `bot_protection` in config.
So I guess I will change it to `access_control`.
This referenced the similar feature of auth0:
https://auth0.com/docs/secure/tenant-access-control-list